metrics_api/test/opentelemetry/metrics/instrument/asynchronous_instrument_test.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Metrics::Instrument::AsynchronousInstrument do
+  describe '#name' do
+    it 'returns name' do
+      instrument = build_asynchronous_instrument('test-instrument')
+
+      assert(instrument.name == 'test-instrument')
+    end
+  end
+
+  describe '#unit' do
+    it 'returns unit' do
+      instrument = build_asynchronous_instrument('test-instrument')
+      assert(instrument.unit == '')
+
+      instrument = build_asynchronous_instrument('test-instrument', unit: 'celsius')
+      assert(instrument.unit == 'celsius')
+    end
+  end
+
+  describe '#description' do
+    it 'returns description' do
+      instrument = build_asynchronous_instrument('test-instrument')
+      assert(instrument.description == '')
+
+      instrument = build_asynchronous_instrument('test-instrument', description: 'room temperature')
+      assert(instrument.description == 'room temperature')
+    end
+  end
+
+  describe '#callbacks' do
+    it 'returns callbacks list' do
+      instrument = build_asynchronous_instrument('test-instrument')
+      assert(instrument.callbacks == [])
+
+      callback_1 = -> {}
+      callback_2 = -> {}
+
+      instrument = build_asynchronous_instrument('test-instrument', callbacks: callback_1)
+      assert(instrument.callbacks == [callback_1])
+
+      instrument = build_asynchronous_instrument('test-instrument', callbacks: [callback_1, callback_2])
+      assert(instrument.callbacks == [callback_1, callback_2])
+    end
+  end
+
+  describe '#register_callbacks' do
+    it 'responds without errors and returns nil' do
+      instrument = build_asynchronous_instrument('test-instrument')
+
+      assert(instrument.register_callbacks(-> {}).nil?)
+      assert(instrument.register_callbacks(-> {}, -> {}).nil?)
+      assert(instrument.register_callbacks([-> {}, -> {}]).nil?)
```

Comment:
```suggestion
      assert(instrument.register_callbacks(-> {}, -> {}).nil?)
```

Reply:
What do you think of allowing all these use cases:

```rb
instr.register_callback(cb1)
instr.register_callback(cb2, cb3)
instr.register_callback([cb4, cb5]) # And support this as well?
# Just so it is consistent with `callbacks:` from the `Meter#create_observable_counter` API
```

See https://github.com/open-telemetry/opentelemetry-ruby/pull/1442#discussion_r1214852101